samples/drivers/rtc_ds3231/README.rst

```diff
@@ -0,0 +1,155 @@
+.. _rtc_ds3231_sample:
+
+DS3231 TCXO RTC Sample Application
+##################################
+
+Overview
+********
+
+The `DS3231`_ temperature-compensated real-time clock is a
+high-precision (2 ppm) battery backed clock that maintains civil time
+and supports alarms.  The `Chronodot`_ breakout board can be used to
+test it.
+
+Annotated Example Output
+************************
+
+The sample first displays the boot banner; board identifier and
+frequency of the local clock used for synchronization; and whether the
+DS3231 has recorded a loss-of-oscillator::
+
+   ***** Booting Zephyr OS build zephyr-v1.14.0-2409-g322d53aedaa0 *****
+   DS3231 on particle_xenon syncclock 1000 Hz
+   .
+   DS3231 has not experienced an oscillator fault
+
+Next information about the device as a counter is displayed.  The
+counter value is read, and its value formatted as the date, time, day of
+week, and day of year (19 July 2019 is a Friday, and is the 200th day of
+2019)::
+
+   Counter at 0x20001a58
+           Max top value: 4294967295 (ffffffff)
+           2 channels
+           1 Hz
+   Top counter value: 4294967295 (ffffffff)
+   Now 1563512509: 2019-07-19 05:01:49 Fri 200
+
+The DS3231 control and status register values are displayed::
+
+   DS3231 ctrl 04 ; ctrl_stat 08
+
+Next, if the sample application option
```

Comment:
If you write this like so, it will link to the config option documentation:
```
Next, if the sample application option :option:`CONFIG_APP_SET_ALIGNED_CLOCK`
is set, the civil time will be advanced to ...
```